Add KF 0.7 on OpenShift docs content

vpavlin opened the pull request:
Adding documentation for deploying Kubeflow 0.7 on OpenShift 4 (specifically tested on 4.2, we'll update it when we re-test on latest 4.3)

This points to our fork of manifest repository, but we plan on proposing the changes to the KF upstream repository and get up to speed with master of the manifests repository, so we will also update that when we have things fixed there.

Resolves: #33 

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/website/1636)
<!-- Reviewable:end -->

joeliedtke commented:
/assign

joeliedtke commented:
/ok-to-test

vpavlin commented:
@joeliedtke I have just realized I made this on top of `master` branch, but since we are talking about 0.7, would it make more sense to send the PR for v0.7-branch? Not sure what the preference is

sarahmaddox commented:
@joeliedtke and @vpavlin I'm hopping in here because we've had a few discussions about this recently.

Our pattern is to create the docs in the master branch always. If the instructions are not relevant to the latest version (in this case v1.0.0) then we should add a note at the top of the doc explaining that this doc has not yet been updated for the latest release.

If we want to include the docs in the v0.7 branch, then we should cherry-pick the commit once it's merged into master.

joeliedtke commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+cd manifests
+```
+
+2. Build the deployment configuration using the Openshift `KFDef` file and local downloaded manifests 
````
Remove code formatting from KfDef. If you feel that this needs to be emphasized, consider bolding it. This is because KfDef refers to a type of file definition, rather than a  command.

vpavlin replied:
Yeah, I know about the list, but somehow I still prefer to keep the numbering in the MD:) But I changed it to keep consistent with your best practices, thanks

joeliedtke commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+
+2. Build the deployment configuration using the Openshift `KFDef` file and local downloaded manifests 
+
+> At the time of writing this document there is Kubeflow [bug](https://github.com/kubeflow/kubeflow/issues/4678) which prevents downloading the manifests during a build process.
```
Also, indent this line with 3 spaces. I think that this line is breaking the Markdown list formatting. You may need to indent the code block as well.

vpavlin replied:
Good point, I also had to indent all the code blocks! Thanks!

joeliedtke commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+4. Wait until all the pods are running
+
+```
+oc get pods
````
Would it be possible to add an example of what this looks like when it is complete? For example:

The result should be something like this:

```
[example of all pods running result]
```

joeliedtke commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+
+```
+oc get routes -n istio-system istio-ingressgateway -o jsonpath='http://{.spec.host}/'
+```
````
Consider adding a Next steps section to suggest activities for users who have just installed Kubeflow

Here is an [example of a next steps section](https://master.kubeflow.org/docs/gke/deploy/deploy-ui/#next-steps) after installing Kubeflow

vpavlin replied:
Added a few next steps items, I'll keep this in mind and will try to add more later

joeliedtke commented on content/docs/openshift/uninstall-kubeflow.md:
```diff
@@ -0,0 +1,17 @@
++++
+title = "Uninstall Kubeflow"
+description = "Instructions for uninstalling Kubeflow"
```
Suggested revision: add that these instructions apply to OpenShift deployments

Instructions for uninstalling Kubeflow from your OpenShift cluster

vpavlin commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+### OpenShift 4 cluster
+
+* You need to have access to an OpenShift 4 cluster as `cluster-admin` to be able to deploy Kubeflow.
+* You can run a local cluster using [Code Ready Containers](https://code-ready.github.io/crc/), use https://try.openshift.com or bring your own cluster.
```
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+
+* You need to have access to an OpenShift 4 cluster as `cluster-admin` to be able to deploy Kubeflow.
+* You can run a local cluster using [Code Ready Containers](https://code-ready.github.io/crc/), use https://try.openshift.com or bring your own cluster.
+* You will need to login as a user with `cluster-admin` privileges to the cluster to be able to deploy Kubeflow.
```
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+* You need to have access to an OpenShift 4 cluster as `cluster-admin` to be able to deploy Kubeflow.
+* You can run a local cluster using [Code Ready Containers](https://code-ready.github.io/crc/), use https://try.openshift.com or bring your own cluster.
+* You will need to login as a user with `cluster-admin` privileges to the cluster to be able to deploy Kubeflow.
+* Install [`oc` command line tool](https://docs.openshift.com/container-platform/4.2/cli_reference/openshift_cli/getting-started-cli.html) to communicate with the cluster.
```
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+
+#### Code Ready Containers
+
+If you are using Code Ready Containers, you will need to make sure you have enough resources configured for the VM:
```
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+Recommended: 
+
+```
+16GB memory
````
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+
+```
+16GB memory
+6 cpu
````
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+```
+16GB memory
+6 cpu
+5G disk space
````
Updated both, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+Minimal:
+
+```
+10GB memory
````
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+
+```
+10GB memory
+6 cpu
````
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+```
+10GB memory
+6 cpu
+30G disk space (default for CRC)
````
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+
+## Installing Kubeflow
+
+To install Kubeflow 0.7 on OpenShift 4.2 please follow the steps below:
```
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
```diff
@@ -0,0 +1,74 @@
+
+To install Kubeflow 0.7 on OpenShift 4.2 please follow the steps below:
+
+1. Clone the [opendatahub/manifests](https://github.com/opendatahub-io/manifests) fork repo which defaults to the branch `v0.7.0-branch-openshift`
```
Updated, thanks

vpavlin commented on content/docs/openshift/install-kubeflo.md:
````diff
@@ -0,0 +1,74 @@
+oc get pods
+```
+
+5. Get a URL and launch the Kubeflow portal
````
I tried to reword it so that it gives more information on what is happening

vpavlin commented:
I incorporated your suggestions, the only thing I cannot do at the moment is adding the output of `oc get pods` because my cluster is down:) I also renamed the install MD file because there was a typo (kubflo vs. kubeflow)

joeliedtke commented on content/docs/openshift/install-kubeflow.md:
```diff
@@ -0,0 +1,82 @@
+### OpenShift 4 cluster
+
+* You need to have access to an OpenShift 4 cluster as `cluster-admin` to be able to deploy Kubeflow.
+* You can use Code Ready Containers](https://code-ready.github.io/crc/) (CRC) to run a local cluster, use [try.openshift.com](https://try.openshift.com) to create a new cluster or use an existing cluster.
```
The Markdown for the Code Ready Containers link is broken. Please update: Code Ready -> [Code Ready

vpavlin replied:
Oh man...sorry..fixed

joeliedtke commented:
These changes can be previewed at:
https://deploy-preview-1636--competent-brattain-de2d6d.netlify.com/docs/openshift/
https://deploy-preview-1636--competent-brattain-de2d6d.netlify.com/docs/openshift/install-kubeflow/
https://deploy-preview-1636--competent-brattain-de2d6d.netlify.com/docs/openshift/uninstall-kubeflow/https://deploy-preview-1636--competent-brattain-de2d6d.netlify.com/docs/started/getting-started/

joeliedtke commented:
Your PR currently has a conflict on `content/docs/started/getting-started.md`. Would you please resolve that conflict?

joeliedtke commented:
/lgtm
/approve

Thanks @vpavlin, this looks great!
